src/cloud/social/alias/randombytes.js

```diff
@@ -1,33 +1,55 @@
 // Alternative to crypto-browserify's randombytes shim which
-// falls back to Math.random when crypto.getRandomValues is
+// falls back to freedom.js core.crypto when crypto.getRandomValues is
 // unavailable, which is currently the case in Firefox web workers.
-// TODO: Fallback to something better such as freedomjs' core.crypto
-//       module (difficult because it is an asynchronous API).
+// TODO: post FF48 (2016-08-02) switch to just native crypto.getRandomValues
 
-// Firefox raises an error if we so much as even try
-// to *access* crypto, hence this bizarre construction.
-var cryptoAvailable = true;
-try {
-  if (crypto) {}
-} catch(e) {
-  console.warn('crypto unavailable, falling back to Math.random');
-  cryptoAvailable = false;
+var nativeCrypto = typeof crypto !== 'undefined';
+var refreshBuffer, bufferRemaining;
+if (!nativeCrypto) {
+  // Filling with freedom.js core.crypto, which requires a buffer due to async
+  var rand = freedom['core.crypto'](),
+      buf,
+      offset = 0;
+  refreshBuffer = function (size) {
+    return rand.getRandomBytes(size).then(function (bytes) {
+      buf = new Uint8Array(bytes);
+      offset = 0;
+    }, function (err) {
+      console.log(err);
+    });
+  }.bind(this);
+  refreshBuffer(10000);  // initial randomness
+  bufferRemaining = 10000;
+
+  crypto = {};
```

Comment:
I think monkey patching `crypto` like this is a little hard to read and ends up less efficient than it could be. Suggest you have a method, e.g. `getBufferedRandomBytes(size)`, which returns a `Buffer` of `size` random bytes. It throws if there are insufficient bytes available and refreshes the buffer if there are less than `size * 10` bytes remaining.